servant/test/Servant/LinksSpec.hs

```diff
@@ -172,8 +172,8 @@ spec = describe "Servant.Links" $ do
 
     it "generated correct links for DeepQuery" $ do
       let bFilter = Proxy :: Proxy ("books" :> DeepQuery "filter" BookQuery :> Get '[JSON] [Book])
-      let exampleQuery = BookQuery { author = "Frank Herbert", year = 1965 }
-      apiLink bFilter exampleQuery `shouldBeLink` "books?filter"
+      let exampleQuery = BookQuery { author = "Herbert", year = 1965 }
+      apiLink bFilter exampleQuery `shouldBeLink` "books?filter[author]=Herbert&filter[year]=1965"
```

Comment:
@clementd-fretlink I'm correct here right? This is what we expect? 👀 
```sh
       expected: "books?filter[author]=Herbert&filter[year]=1965"
        but got: "books?filter%5Bauthor%5D=Herbert&filter%5Byear%5D=1965"
```

Reply:
yes, square brackets are url-encoded to %5B and %5D. Brackets still work when put in a URL, but since the `HasLink` instance url-encodes fragments (as it should, imo) we get this result

Reply:
Awesome then I'll update the text to reflect that, I thought it was weird that it returned the encoded square brackets  😄